fastly/fastly_test_utils.go

```diff
@@ -666,3 +667,45 @@ Wm7DCfrPNGVwFWUQOmsPue9rZBgO
 -----END CERTIFICATE-----
 `
 }
+
+type FunctionalTest struct {
```

Comment:
Great point here.

Reply:
I don't but I wasn't in a position start refactoring all of the stuff in the `fastly` package in this PR. I can easily move these two new pieces, though, I think.

Reply:
Also, the import cycle you supposed might happen has already happened; one of the commits in this PR moves `image_optimizer_default_settings_test` to its own package because it needs to import `fastly/products/image_optimizer` (new package in this PR), which imports `fastly`, so a cycle existed.

Reply:
These have been moved.